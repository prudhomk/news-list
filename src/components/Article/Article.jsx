import React from 'react';
import PropTypes from 'prop-types';

const Article = ({ title, author, description }) => (
  <li>
    <p>{title}</p>
    <p>{author}</p>
    <p>{description}</p>
  </li>
);

Article.propTypes = {
  title: PropTypes.string.isRequired,
  author: PropTypes.string.isRequired,
  description: PropTypes.string.isRequried,
};

export default Article;
