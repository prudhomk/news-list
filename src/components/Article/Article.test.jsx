import React from 'react';
import { render } from '@testing-library/react';
import Article from './Article';

describe('Article Component', () => {
  it('renders article', () => {
    const { asFragment } = render(
      <Article
        title="Race to the Outhouse"
        author="Will E. Makeit"
        description="a sad tale as old as time"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
