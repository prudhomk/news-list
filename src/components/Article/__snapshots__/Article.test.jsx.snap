// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article Component renders article 1`] = `
<DocumentFragment>
  <li>
    <p>
      Race to the Outhouse
    </p>
    <p>
      Will E. Makeit
    </p>
    <p>
      a sad tale as old as time
    </p>
  </li>
</DocumentFragment>
`;
